yellowbrick/cluster/silhouette.py

```diff
@@ -37,11 +37,65 @@
 
 class SilhouetteVisualizer(ClusteringScoreVisualizer):
     """
-    The Silhouette Visualizer displays the silhouette coefficient for each 
-    sample on a per-cluster basis, visualizing which clusters are dense and 
-    which are not. This is particularly useful for determining cluster 
-    imbalance, or for selecting a value for K by comparing multiple 
+    The Silhouette Visualizer displays the silhouette coefficient for each
+    sample on a per-cluster basis, visually evaluating the density and
+    separation between clusters. The score is calculated by averaging the
+    silhouette coefficient for each sample, computed as the difference
+    between the average intra-cluster distance and the mean nearest-cluster
+    distance for each sample, normalized by the maximum value. This produces a
+    score between -1 and +1, where scores near +1 indicate high separation
+    and scores near -1 indicate that the samples may have been assigned to
+    the wrong cluster.
+
+    In SilhouetteVisualizer plots, clusters with higher scores have wider
+    silhouettes, but clusters that are less cohesive will fall short of the
+    average score across all clusters, which is plotted as a vertical dotted
+    red line.
+
+    This is particularly useful for determining cluster
+    imbalance, or for selecting a value for K by comparing multiple
     visualizers.
+
+    Parameters
+    ----------
+    model : a Scikit-Learn clusterer
+        Should be an instance of a centroidal clustering algorithm (``KMeans``
+        or ``MiniBatchKMeans``).
+
+    ax : matplotlib Axes, default: None
+        The axes to plot the figure on. If None is passed in the current axes
+        will be used (or generated if required).
+
+    kwargs : dict
+        Keyword arguments that are passed to the base class and may influence
+        the visualization as defined in other Visualizers.
+
+    Attributes
+    ----------
+    silhouette_score_ : float
+        Mean Silhouette Coefficient for all samples. Computed via scikit-learn
+        `sklearn.metrics.silhouette_score`.
+
+    silhouette_samples_ : array, shape = [n_samples]
+        Silhouette Coefficient for each samples. Computed via scikit-learn
+        `sklearn.metrics.silhouette_samples`.
+
+    n_samples : integer
+        Number of total samples in the dataset (X.shape[0])
+
+    n_clusters : integer
+        Number of clusters (e.g. n_clusters or k value) passed to internal
+        scikit-learn model.
+
```

Comment:
`n_samples` and `n_clusters` aren't suffixed with an `_` -- and that's my bad; they should be since they're both computed in `fit()` and also the `# Required internal properties` part of the `__init__` function should be removed so that we can correctly determine if the estimator was fitted or not using the [`sklearn.validation.check_is_fitted`](http://scikit-learn.org/stable/modules/generated/sklearn.utils.validation.check_is_fitted.html) methodology (e.g. these attributes should not be members of the class until after `fit()` is called. 

Thank you so much for adding the docstring because it allows me to review the code that I haven't looked at in a while! I'm now remembering that `SilhouetteVisualizer` was a prototype and still had a lot of work remaining (e.g. in #516). I'd appreciate it if you wouldn't mind deleting lines 109-113 and adding the `_` suffix or adding this to the tasks to complete in #516.

Reply:
You got it!

Reply:
Thanks for taking care of my technical debt!